source/src/protocols/motifs/LigandDiscoverySearch.cc

```diff
@@ -292,694 +366,1790 @@ utility::vector1<core::conformation::ResidueOP> LigandDiscoverySearch::get_all_r
+
+		for ( const auto & best_pose_runner : best_placements ) {
+
+			ms_tr.Debug << "Making pdb file for: " << std::get<2>(best_pose_runner) << std::endl;
+
+			core::io::pdb::dump_pdb( ( std::get<1>(best_pose_runner)), std::get<2>(best_pose_runner));
+		}
+
+
+		ms_tr.Debug << "Number of placements that passed all filters: " << passed_placement_counter << std::endl;
+
+	}
+	return 1;
+}
+
+// @brief A function to reset the working_pose_ back to its original state, as found in original_pose
+//to call this function the least amount of times (and minimize wasted time/resources), this function is only to be called right before a placement is culled before a continue statement after working_pose_ had a ligand added to it
+void LigandDiscoverySearch::reset_working_pose()
+{
+	core::pose::PoseOP original_poseop(original_pose_.clone());
+	//assign the original_poseop to working_pose_
+	working_pose_ = original_poseop;
+}
+
+//function to get a sub-library of motifs from the main library, based on the residue being used (only get for select residue)
+//function may have use outside discover, so public can use it
+protocols::motifs::MotifCOPs LigandDiscoverySearch::get_motif_sublibrary_by_aa(std::string residue_name)
+{
+	static basic::Tracer ms_tr( "LigandDiscoverySearch.get_motif_sublibrary_by_aa", basic::t_info );
+
+	//create temporary motifcops object to hold
+	protocols::motifs::MotifCOPs motif_holder;
+
+	for ( auto motifcop : motif_library_ ) {
+		//collect residue name from motifcop
+		std::string motif_residue_name(motifcop->restype_name1());
+
+		//if residue name matches that of the residue at the working position, add it to the motif_holder
+		if ( motif_residue_name == residue_name ) {
+			motif_holder.push_back(motifcop);
+		}
+	}
+
+	//assign motif_holder contents to motif_library_for_select_residue_
+	motif_library_for_select_residue_ = motif_holder;
+
+
+	ms_tr.Debug << "Created motif sub-library for residue " << residue_name << " with " << motif_library_for_select_residue_.size() << " motifs in it." << std::endl;
+
+
+	return motif_holder;
+}
+
+//create protein_representation_matrix_
+//uses working_pose to make the matrix
+void LigandDiscoverySearch::create_protein_representation_matrix(core::Size & x_shift, core::Size & y_shift, core::Size & z_shift, int & x_bound_int, int & y_bound_int, int & z_bound_int)
+{
+
+	//create tracer to identify points of the run
+	static basic::Tracer ms_tr( "LigandDiscoverySearch.create_protein_representation_matrix", basic::t_info );
+
+	//run through all atoms to derive a range of dimensions to contain the protein in a 3D  space
+	//since we can't have negative indices, we need to normalize the coordinate values so that everything is positive
+	//derive constant values based  on the most negative values in each dimension, and then add that constant to all coordinates
+
+	int smallest_x = 1;
+	int smallest_y = 1;
+	int smallest_z = 1;
+
+	int largest_x = 1;
+	int largest_y = 1;
+	int largest_z = 1;
+
```

Comment:
Just a note that PDBs aren't guaranteed to be centered around 0. You can theoretically have a protein structure which has z coordinates which range from -501 to -554, or x coordinates which range from 356 to 467. Starting with each at 1 might not be what you want.